R/wordbankr.R

```diff
@@ -4,10 +4,69 @@ utils::globalVariables(c(".", "n"))
 #' @importFrom rlang .data
 NULL
 
+#' Locate Wordbank Database
+#'
+#' @param mode A string indicating connection mode: one of \code{"local"},
+#'   \code{"remote"} (defaults to \code{"remote"}), or a date given in
+#'   \code{"YYYYMMDD"} or \code{"MMDDYYY"} format.
+#' @return A \code{src} object which is a connection to the Wordbank database.
+#' @keywords internal
+#'
+#' @examples
+#' \dontrun{
+#' db_info <- find_database("2017-03-22")
+#' }
+find_database <- function(mode = "remote") {
+  db_info = list()
+  if (mode == "remote") {
+    db_info[c("address", "dbname")] = c("server.wordbank.stanford.edu", "wordbank")
+  } else if (mode == "local") {
+    db_info[c("address", "dbname")] = c("localhost", "wordbank")
+  } else {
+    db_info["address"] <- "wordbank-backup.canyiscnpddk.us-west-2.rds.amazonaws.com"
+    mode_date <- tryCatch(lubridate::parse_date_time(mode, orders = c("mdY", "Ymd")),
+                          warning = function(e) {
+                            message("Error parsing date entered in mode variable.")
+                            message("Please enter dates as in 'Ymd' or 'mdY' format.")
+                            message("Grabbing latest archived version of Wordbank.")
+                            lubridate::today()})
+      
+    my_host <- DBI::dbConnect(RMySQL::MySQL(),
+                   host = db_info$address,
+                   user = "wordbank", password = "wordbank")
+    
+    if (mode_date < lubridate::parse_date_time("2017-01-10", "Ymd")) {
+      message("Entered date is earlier than the first archived version of Wordbank.")
+      message("Referring to earliest version of Wordbank archived 2017-01-10.")
+      
+      db_list <- DBI::dbGetQuery(my_host, "show databases;") %>%
+        dplyr::collect() %>%
+        dplyr::filter(grepl('wordbank_[0-9]+', .data$Database)) %>%
+        dplyr::mutate(date_archived = lubridate::parse_date_time(gsub("wordbank_", "", .data$Database), "Ymd")) %>%
+        dplyr::filter(.data$date_archived == min(.data$date_archived))
+      
+    } else{
+      
+      db_list <- DBI::dbGetQuery(my_host, "show databases;") %>%
+        dplyr::collect() %>%
+        dplyr::filter(grepl('wordbank_[0-9]+', .data$Database)) %>%
+        dplyr::mutate(date_archived = lubridate::parse_date_time(gsub("wordbank_", "", .data$Database), "Ymd")) %>%
+        dplyr::filter(.data$date_archived <= mode_date) %>%
```

Comment:
it should be clearer that the date supplied doesn't have to be exact and that what's retrieved is the latest version that's before or on that date – maybe explain that in the documentation and show a message with exactly with version ends up being retrieved